### PR TITLE
pause: encode Windows OS version in image tag instead of repository

### LINKED ISSUE #97780: "repository name must be lowercase" when building pause image for Windows 20H2
```
error: invalid tag "$REGISTRY/pause-windows-amd64-20H2:3.4": repository name must be lowercase
```

We could either:
1. `echo ${OSVERSION} | tr [:upper:] [:lower:]` where appropriate - least amount of code change but cumbersome
2. Encode OS version in the tag instead of the repository

/cc @claudiubelu

### PR DESCRIPTION
**Pull request by chewong**
Signed-off-by: Ernest Wong <chuwon@microsoft.com>

**What type of PR is this?**

Fixes #97780 

cloud build output: https://pastebin.ubuntu.com/p/QgrFfpW97k/
`docker manifest inspect`: https://pastebin.ubuntu.com/p/ZZW9yjhk6t/

/sig windows
/kind cleanup
/assign @claudiubelu 

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by claudiubelu** [comment]
FWIW, indeed, this PR fixes the issue. LGTM!

File: build/pause/Makefile [hunk 1]
```diff
@@ -16,7 +16,6 @@
 
 REGISTRY ?= staging-k8s.gcr.io
 IMAGE = $(REGISTRY)/pause
-IMAGE_WITH_OS_ARCH = $(IMAGE)-$(OS)-$(ARCH)
 
 TAG = 3.4
```
Comment (justaugustus) [inline build/pause/Makefile:20]:
```suggestion
TAG = 3.4.1
```

**Review by justaugustus — CHANGES_REQUESTED** [review]
LGTM, but let's bump the tag here as well.

**Pull request merged**